core/templates/dev/head/components/ck-editor-helpers/ck-editor-5-rte.directive.ts

```diff
@@ -88,17 +88,76 @@ angular.module('oppia').directive('ckEditor5Rte', [
           scope.uiConfig().startupFocusEnabled !== undefined) {
           startupFocusEnabled = scope.uiConfig().startupFocusEnabled;
         }
+        // A RegExp for matching rich text components.
+        var componentRegExp = (
+          /(<(oppia-noninteractive-(.+?))\b[^>]*>)[\s\S]*?<\/\2>/g
+        );
+        var wrapComponents = function(html) {
+          if (html === undefined) {
+            return html;
+          }
+          // This wraps the custom plugin data currently, we are using only
```

Comment:
I don't understand this comment. Can you elaborate please? What does "default toolbar" have to do with it?

Reply:
Hi @seanlip, default toolbar means toolbar with no custom plugins.
This function is used to wrap the plugin data with needed tags, classes or built directives like for inline we wrap it with `span` and for long inputs we wrap it with containers (This can be changed no such rules).

Reply:
Thanks. I understand the term "default toolbar", and I also understand what the function is doing. But I don't understand how the two concepts go together. Aren't they separate things?

I.e. what does this return statement have to do with the "default toolbar"? 

In any case, perhaps you could change the comment to say something like "This wraps the custom RTE components in regular spans/divs, so that XXX." (I'm not sure what XXX is, that part is something you should fill in clearly. Try to explain what would happen if we didn't wrap it.)

Reply:
Hi @DubeySandeep & @YashJipkate, if you are free PTAL here.

What the wrapcomponent function does is, it wraps the oppia's custom plugin(Like link,Mathjax etc) data with Ckeditor allowed tags (see [here](https://ckeditor.com/docs/ckeditor5/latest/builds/guides/faq.html#why-does-the-editor-filter-out-my-content-styles-classes-elements-where-is-configallowedcontent-true))

The output data for of the custom link plugin will be like "<oppia-noninteractive-link><a href='link></a></oppia-noninteractive-link>" which cannot be directly set to CK so we need to wrap it with the span tags (because 'a' is an inline tag for other block tags we wrap by divs)

See the below image :

<img width="823" alt="Screenshot 2019-08-12 at 10 33 21 PM" src="https://user-images.githubusercontent.com/41017303/62883959-ffe9ef80-bd52-11e9-85ac-7020fc012f02.png">

As wrapping is needed for custom plugin data with this "We currently use default toolbar" in the comments,  I meant that we are using the default toolbar so if we wrap or don't wrap it won't affect the CKEditor
<img width="348" alt="Screenshot 2019-08-12 at 10 31 58 PM" src="https://user-images.githubusercontent.com/41017303/62884278-c06fd300-bd53-11e9-987b-9b285ee3eb58.png">


though I have added this function in this PR as I was making the data-flow and it was mentioned in the ck-5 docs and CKEditor-4 to have a valid wrap.

Reply:
Why do you say "the output data for of the custom link plugin will be like `"<a href='link>"`"? I thought the noninteractive link plugins would be of the form oppia-noninteractive-link etc.

For the comment, perhaps you could go with something like this then (assuming it's correct):

```
          // This wraps the custom RTE components in regular spans/divs,
          // as they are allowed wrap tags in CKEditor. This avoids data loss while
          // setting data to ckeditor.
          // See https://ckeditor.com/docs/ckeditor5/latest/builds/guides/faq.html
          // for more details (under the section "Why does the editor filter out my ...").
```

Reply:
Hi @seanlip, yes you are correct it will be a noninteractive link. in my previous comment, I used this tag `<oppia-noninteractive-link> ` without the insert code option so it didn't render in preview

Reply:
Ah I see, thanks, make sure to always check your previews after writing comments in GitHub. Will take a look at the PR!